boutdata/tests/test_collect.py

```diff
@@ -0,0 +1,2773 @@
+from glob import glob
+import numpy as np
+import numpy.testing as npt
+from pathlib import Path
+import pytest
+
+from boutdata.collect import collect
+from boutdata.squashoutput import squashoutput
+
+from boutdata.tests.make_test_data import (
+    apply_slices,
+    create_dump_file,
+    concatenate_data,
+    expected_attributes,
+    remove_xboundaries,
+    remove_yboundaries,
+    remove_yboundaries_upper_divertor,
+)
+
+# Note - using tmp_path fixture requires pytest>=3.9.0
+
+collect_kwargs_list = [
+    {"xguards": True, "yguards": "include_upper"},
+    {"xguards": False, "yguards": "include_upper"},
+    {"xguards": True, "yguards": True},
+    {"xguards": False, "yguards": True},
+    {"xguards": True, "yguards": False},
+    {"xguards": False, "yguards": False},
+]
+
+
+def check_collected_data(
+    expected,
+    *,
+    fieldperp_global_yind,
+    doublenull,
+    path,
+    squash,
+    collect_kwargs,
+    squash_kwargs={},
+):
+    # Apply effect of arguments to expected data
+    if not collect_kwargs["xguards"]:
+        remove_xboundaries(expected, expected["MXG"])
+    if collect_kwargs["yguards"] is True and doublenull:
+        remove_yboundaries_upper_divertor(
+            expected, expected["MYG"], expected["ny_inner"]
+        )
+    if not collect_kwargs["yguards"]:
+        remove_yboundaries(expected, expected["MYG"], expected["ny_inner"], doublenull)
+
+    collect_kwargs = collect_kwargs.copy()
+    if squash:
+        squashoutput(path, outputname="boutdata.nc", **collect_kwargs)
+        collect_kwargs["prefix"] = "boutdata"
+        # Delete dump files to be sure we do not read from them
+        dump_names = glob(str(path.joinpath("BOUT.dmp.*.nc")))
+        for x in dump_names:
+            Path(x).unlink()
```

Comment:
I guess this is ok when running tests in parallel as the `tmp_path` fixture is being used?

Reply:
Yes, quoting the docs https://docs.pytest.org/en/6.2.x/tmpdir.html: "the tmp_path fixture which will provide a temporary directory unique to the test invocation, created in the base temporary directory."